### PR TITLE
RUMM-670 Fix the active span concept to let create several root spans on the same thread

### PR DESCRIPTION
**Pull request by nachoBonafonte**
### What and why?

Fix the active span concept to let create several root spans on the same thread. Previous to this change with this code:

````
let job1Span = tracer.startSpan("async work 1”)
doSomethingAsynchronously(completion: { job1Span.finish() })

let job2Span = tracer.startSpan("async work 2")
doSomethingAsynchronously(completion: { job2Span.finish() })
````

`job2Span` would be assigned as child of` job1Span`

With this change both spans will have the same parent (the `activeSpan` or none if there is no `activeSpan`). 
For job2Span to be child of job1span you must now set it explicitly as its parent:
````
let job1Span = tracer.startSpan("async work 1”)
doSomethingAsynchronously(completion: { job1Span.finish() })

let job2Span = tracer.startSpan("async work 2", childOf: job1Span)
doSomethingAsynchronously(completion: { job2Span.finish() })
````
or you must set `job1Span` as active span before creating `job2Span`:
````
let job1Span = tracer.startSpan("async work 1”)
doSomethingAsynchronously(completion: { job1Span.finish() })
job1Span.setActive()

let job2Span = tracer.startSpan("async work 2")
doSomethingAsynchronously(completion: { job2Span.finish() })
````
### How?

Now, a span is not automatically set as the `activeSpan` when it is created, there is a new method `setActive()` that must be called explicitly in the span for it.
Any span will still use the current `activeSpan` as its parent on creation, except if any other explicit parent is assigned.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

**Comment by ncreated** [comment]
> i can't think of a good name though.. (i can only come up with postmodern-sounding names like setContextualRootSpan())

"Active Span" is a common term defined as an [optional element in the Open Tracing API](https://github.com/opentracing/specification/blob/master/specification.md#optional-api-elements), we shouldn't change its name 😉.

File: Sources/Datadog/OpenTracing/OTTracer.swift [hunk 1]
```diff
@@ -57,6 +57,8 @@ public protocol OTTracer {
 
     /// Returns the active span in the current execution context, this span will be automatically assigned as the parent of any
     /// created Span without explicit parent.
+    /// For a span to be set as the active one in the current context,  it must explicitly call span.setActive()
+    /// This method can return different results depending on the context that is executed.
```
Comment (ncreated) [inline Sources/Datadog/OpenTracing/OTTracer.swift:61]:
```suggestion
    /// For a span to be set as the active one in the current context, it must explicitly call `span.setActive()`.
    /// This method can return different results depending on the context that is executed.
```

File: Sources/Datadog/Tracing/DDSpan.swift [hunk 2]
```diff
@@ -113,10 +107,24 @@ internal class DDSpan: OTSpan {
...
 
+    @discardableResult
+    func setActive() -> OTSpan {
+        let dso = UnsafeMutableRawPointer(mutating: #dsohandle)
+        let activity = _os_activity_create(dso, "InitDDSpanContext", OS_ACTIVITY_CURRENT, OS_ACTIVITY_FLAG_DEFAULT)
+        activityId = os_activity_get_identifier(activity, nil)
+        os_activity_scope_enter(activity, &activityState)
+        if let activityId = activityId {
+            ddTracer.activeSpansPool.addSpan(span: self, activityId: activityId)
+        }
```
[the remark this answers is not in the document]
Comment (nachoBonafonte) [reply Sources/Datadog/Tracing/DDSpan.swift:124]: Ok, sounds good, I will work on this

**Comment by nachoBonafonte** [comment]
I have added a test that call setActive multiple times, to check it works as expected.

**Pull request merged**